src/libs/IOUUtils.ts

```diff
@@ -141,10 +141,11 @@ function insertTagIntoTransactionTagsString(transactionTags: string, tag: string
     const tagArray = TransactionUtils.getTagArrayFromName(transactionTags);
     tagArray[tagIndex] = tag;
 
-    return tagArray
-        .map((tagItem) => tagItem.trim())
-        .filter((tagItem) => !!tagItem)
-        .join(CONST.COLON);
+    while (tagArray.length > 0 && !tagArray[tagArray.length - 1]) {
+        tagArray.pop();
+    }
+
+    return tagArray.map((tagItem) => tagItem.trim()).join(CONST.COLON);
```

Comment:
@daledah will only the end values be falsey? the original code removes falsy from anywhere in the array. Your tests don't cover a case with falsey in the middle. I don't have full context on tags cc @situchan

Reply:
Unfortunately, we cannot remove falsey values in the middle. It causes regression (array index mismatch).
This was original code (before [regression PR](https://github.com/Expensify/App/pull/46977)) which trims only last space after `:`:
```tagArray.join(CONST.COLON).replace(/:*$/, '')```

Reply:
thanks for the context @situchan